File: 2022-05/javascript/19_Remove_Nth_Node_From_End_of_List.js
function removeNthFronEnd(head, n) {
    const start = new ListNode(0, head)
    let slow = start, fast = start
    while (n >= 0) {
        n--
        fast = fast.next
    }
    while (fast) {
        slow = slow.next
        fast = fast.next
    }
    slow.next = slow.next.next
    return start.next
}

File: 2022-05/python/19_Remove_Nth_Node_From_End_of_List.py
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        start = ListNode(0, head)
        slow, fast = start, start
        while n >= 0:
            n -= 1
            fast = fast.next
        while fast:
            fast.next
            slow.next
        slow.next = slow.next.next
        return start.next